### ISSUE TITLE
CivetWeb Enable WebSocket

### ISSUE THREAD

**Issue by Nukersson** [zephyrproject-rtos/zephyr#27417]
**Is your enhancement proposal related to a problem? Please describe.**
Compiling **CivetWeb** for `nucleo_f767zi` board with `WebSocket` support is not possible.

Steps to reproduce are:
1. In `zephyrproject/modules/lib/civetweb/CMakeLists.txt` enable the option `CIVETWEB_ENABLE_WEBSOCKETS`
2. Compile the project as follows: `west build -p auto -b "nucleo_f767zi" "samples/net/sockets/civetweb"`
3. Following error should occur:
```
<some_your_path>/zephyrproject/modules/lib/civetweb/src/civetweb.c: In function 'websocket_client_thread':
<some_your_path>/zephyrproject/modules/lib/civetweb/src/civetweb.c:17311:19: error: storage size of 'sa' isn't known 17311 |  struct sigaction sa;
```

It seems to be, that there is no `signal.h` implementation for any `stm32`!

**Describe the solution you'd like**
Clear and proper project compilation with `CIVETWEB_ENABLE_WEBSOCKETS` option.

The file `<some_your_path>/zephyrproject/modules/hal/esp-idf/components/newlib/include/sys/signal.h` seems to have necessary structure. 

**Additional context**
According to above info I have following questions:
* Does anybody already up to integration of WebSockets within CivetWeb to Zephyr and/or stm32?
* Is it possible and/or necessary to adapt `<some_your_path>/zephyrproject/modules/hal/esp-idf/components/newlib/include/sys/signal.h` for stm32?

**Comment by Nukersson** [comment]
@jukkar @erwango @ABOSTM Any ideas or references to CivetWeb Websocket integration group?

**Comment by Nukersson** [comment]
@jukkar @erwango @ABOSTM I found a way to compile my currently under development `websocket_server` sample with CivetWeb for Zephyr. Please follow on [this link](https://github.com/zephyrproject-rtos/civetweb/pull/2) to see the solution.

**Comment by Nukersson** [comment]
Can be solved by e5e0dbbe1b69a88f41ae0f4123cd48eed1238340. It was already added by [following PR](https://github.com/zephyrproject-rtos/civetweb/pull/2)
